src/types.jl

```diff
@@ -17,26 +17,20 @@ struct ManagedSCSMatrix
     values::Vector{Cdouble}
     rowval::Vector{Int}
     colptr::Vector{Int}
-    m::Int
-    n::Int
-end
+    scsmat::Base.RefValue{SCSMatrix}
```

Comment:
RefValue isn't mentioned anywhere in the documentation (https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/), are we supposed to be using this type?

Reply:
it's the return value of `Base.cconvert(Ref{A}, A)`;
Note that we need to ccall with a `Ref` to `SCSData` which contains a `Ptr{SCSMatrix}`. All of the examples given in the docs pass c-compatible julia structs.

However, its just a way of encapsulating the object and its C-compatible reference for `GC.@protect`ed calls. Maybe the complication isn't worth it.

Reply:
It's ok to leave it in, but for the benefit of others reading the code, could you add comments explaining why we need to use an undocumented type here?